lib/spack/spack/directory_layout.py

```diff
@@ -260,9 +260,10 @@ def check_installed(self, spec):
             return None
 
         if not os.path.isfile(spec_file_path):
-            raise InconsistentInstallDirectoryError(
-                'Install prefix exists but contains no spec.yaml:',
-                "  " + path)
+            return None
+#            raise InconsistentInstallDirectoryError(
+#                'Install prefix exists but contains no spec.yaml:',
+#                "  " + path)
```

Comment:
Why exactly is this change necessary? Is it only necessary when running `spack setup`? Could we perhaps write a `spec.yaml` for packages installed by `spack setup`?

I think this check needs to stay here for most package installations (via `spack install`).